src/Analyser/Scope.php

```diff
@@ -1447,10 +1447,10 @@ public function removeTypeFromExpression(Expr $expr, Type $type): self
 		);
 	}
 
-	public function lookForTypeSpecifications(Node $node): self
+	public function lookForTypeSpecifications(Node $node, bool $negated = false): self
```

Comment:
You are right, the bool parameter is quite confusing.

But the original method names confuse me as well. How do you feel about `filterByTruthyValue` + `filterByFalseyValue` or maybe `filterByPositiveCondition` + `filterByNegativeCondition`?